// ng2-alert-service/dist/bootstrap-alert.service.js
import { Injectable } from '@angular/core';
import { Subject } from 'rxjs/Subject';
var BootstrapAlertService = /** @class */ (function () {
    function BootstrapAlertService() {
        this.messageId = 0;
        this.alertEvent = new Subject();
    }
    BootstrapAlertService.prototype.getAlertEvent = function () {
        return this.alertEvent.asObservable();
    };
    BootstrapAlertService.prototype.showSucccess = function (message) {
        this.showToast("success", message);
    };
    BootstrapAlertService.prototype.showInfo = function (message) {
        this.showToast("info", message);
    };
    BootstrapAlertService.prototype.showWarning = function (message) {
        this.showToast("warning", message);
    };
    BootstrapAlertService.prototype.showError = function (message) {
        this.showToast("danger", message);
    };
    BootstrapAlertService.prototype.showToast = function (level, message) {
        var toast = {
            id: this.messageId++,
            type: level,
            message: message,
            messageLocalized: ""
        };
        this.alertEvent.next(toast);
    };
    BootstrapAlertService.decorators = [
        { type: Injectable },
    ];
    /** @nocollapse */
    BootstrapAlertService.ctorParameters = function () { return []; };
    return BootstrapAlertService;
}());
export { BootstrapAlertService };
