// ng4-alert/bundles/ng4-alert.umd.js
(function (global, factory) {
	typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/common'), require('@angular/platform-browser/animations'), require('@angular/animations'), require('rxjs/Subject')) :
	typeof define === 'function' && define.amd ? define(['exports', '@angular/core', '@angular/common', '@angular/platform-browser/animations', '@angular/animations', 'rxjs/Subject'], factory) :
	(factory((global['ng4-alert'] = {}),global.ng.core,global.ng.common,global.ng.platformBrowser.animations,global.ng.animations,global.Rx));
}(this, (function (exports,core,common,animations,animations$1,Subject) { 'use strict';

var Ng4AlertService = (function () {
    function Ng4AlertService() {
        this.ngnSubject = new Subject.Subject();
        this.options = {};
    }
    /**
     * @param {?} options
     * @return {?}
     */
    Ng4AlertService.prototype.ng4Activate = function (options) {
        this.options = options;
        this.options.state = 'active';
        this.ngnSubject.next(options);
    };
    /**
     * @return {?}
     */
    Ng4AlertService.prototype.ng4Deactivate = function () {
        this.options.state = 'inactive';
        this.ngnSubject.next(this.options);
        //return this.ngnSubject.asObservable()
    };
    /**
     * @return {?}
     */
    Ng4AlertService.prototype.ng4State = function () {
        return this.ngnSubject.asObservable();
    };
    return Ng4AlertService;
}());
Ng4AlertService.decorators = [
    { type: core.Injectable },
];
/**
 * @nocollapse
 */
Ng4AlertService.ctorParameters = function () { return []; };
var Ng4AlertComponent = (function () {
    /**
     * @param {?} ng4AlertService
     */
    function Ng4AlertComponent(ng4AlertService) {
        this.ng4AlertService = ng4AlertService;
        this.state = 'inactive';
    }
    /**
     * @return {?}
     */
    Ng4AlertComponent.prototype.ngOnInit = function () {
        var _this = this;
        this.ng4AlertService.ng4State().subscribe(function (message) {
            _this.options = message;
            _this.setState(_this.options.state);
            if (_this.options.autoDismis && _this.options.timeout) {
                setTimeout(function () {
                    _this.setState('inactive');
                }, _this.options.timeout);
            }
            else if (_this.options.autoDismis !== false) {
                setTimeout(function () {
                    _this.setState('inactive');
                }, 2000);
            }
        });
    };
    /**
     * @param {?} state
     * @return {?}
     */
    Ng4AlertComponent.prototype.setState = function (state$$1) {
        this.state = state$$1;
    };
    /**
     * @return {?}
     */
    Ng4AlertComponent.prototype.ngnClose = function () {
        this.ng4AlertService.ng4Deactivate();
    };
    return Ng4AlertComponent;
}());
Ng4AlertComponent.decorators = [
    { type: core.Component, args: [{
                selector: 'ng4-alert',
                template: "\n   <div class=\"ngn-wrapper\" [@ngnState]=\"state\" [ngSwitch]=\"options.type\">\n   \t<div class=\"ngn-alert ngn-default ngn-success\" *ngSwitchCase=\"'success'\">\n   \t\t<div class=\"ngn-icon\"><img src=\"../assets/ic_done_white_24px.svg\"/></div>\n   \t\t<div class=\"ngn-content\">{{options.text}}</div>\n   \t\t<div class=\"ngn-icon ngn-close\" (click)=\"ngnClose()\"><img src=\"../assets/ic_clear_white_18px.svg\"/></div>\n   \t</div>\n   \t<div class=\"ngn-alert ngn-default ngn-danger\" *ngSwitchCase=\"'fail'\">\n   \t\t<div class=\"ngn-icon\"><img src=\"../assets/ic_error_white_24px.svg\"/></div>\n   \t\t<div class=\"ngn-content\">{{options.text}}</div>\n   \t\t<div class=\"ngn-icon ngn-close\" (click)=\"ngnClose()\"><img src=\"../assets/ic_clear_white_18px.svg\"/></div>\n   \t</div>\n   \t<div class=\"ngn-alert ngn-default ngn-warning\" *ngSwitchCase=\"'warning'\">\n   \t\t<div class=\"ngn-icon\"><img src=\"../assets/ic_warning_white_24px.svg\"/></div>\n   \t\t<div class=\"ngn-content\">{{options.text}}</div>\n   \t\t<div class=\"ngn-icon ngn-close\" (click)=\"ngnClose()\"><img src=\"../assets/ic_clear_white_18px.svg\"/></div>\n   \t</div>\n   \t<div class=\"ngn-loader\" *ngSwitchCase=\"'loading'\">\n   \t\t<div class=\"loader\"></div>\n   \t</div>\n   </div>\n\t",
                styles: ["\n   h1 {\n     color: red;\n   }\n\n   .ngn-default{\n       padding: .50rem .8rem;\n       border: 1px solid transparent;\n       border-radius: .25rem;\n   }\n   .ngn-success{\n   \tcolor: #fff;\n       background-color: #5cb85c;\n       border-color: #5cb85c;\n   }\n   .ngn-warning{\n   \tcolor: #fff;\n       background-color: #ec971f;\n       border-color: #eb9316;\n   }    \n\n   .ngn-danger{\n   \tcolor: #fff;\n       background-color: #c9302c;\n       border-color: #c12e2a;\n   }\n   .ngn-icon{\n       display: inline;\n   }\n   .ngn-content{\n       display: inline;\n       vertical-align: super;\n       margin-left: 5px;\n   }\n   /* .ngn-default .ngn-icon img{\n   \tmargin-top: 2px;\n   }\t */\n   .ngn-close{\n   \tfloat: right;\n       cursor: pointer; \n       top: 5px;\n       position: relative;\n       right: -9px;\n       opacity: .5;\n   }\n   .ngn-close:hover{\n   \topacity: 0.7;\n   }\n   .ngn-wrapper{\n       max-width: 200px;\n       min-width: 200px;\n       position: absolute;\n       top: 10px;\n       right: 10px;\n       z-index: 1000;\n   }\n\t"],
                animations: [
                    animations$1.trigger('ngnState', [
                        animations$1.state('inactive', animations$1.style({
                            display: 'none',
                            opacity: 0
                        })),
                        animations$1.state('active', animations$1.style({
                            display: 'block',
                            opacity: 0.7
                        })),
                        animations$1.transition('active <=> inactive', animations$1.animate('300ms')),
                    ])
                ]
            },] },
];
/**
 * @nocollapse
 */
Ng4AlertComponent.ctorParameters = function () { return [
    { type: Ng4AlertService, },
]; };
Ng4AlertComponent.propDecorators = {
    'options': [{ type: core.Input },],
};
var Ng4AlertModule = (function () {
    function Ng4AlertModule() {
    }
    /**
     * @return {?}
     */
    Ng4AlertModule.forRoot = function () {
        return {
            ngModule: Ng4AlertModule,
            providers: [Ng4AlertService]
        };
    };
    return Ng4AlertModule;
}());
Ng4AlertModule.decorators = [
    { type: core.NgModule, args: [{
                imports: [
                    common.CommonModule,
                    animations.BrowserAnimationsModule
                ],
                declarations: [
                    Ng4AlertComponent
                ],
                exports: [
                    Ng4AlertComponent
                ]
            },] },
];
/**
 * @nocollapse
 */
Ng4AlertModule.ctorParameters = function () { return []; };

exports.Ng4AlertModule = Ng4AlertModule;
exports.Ng4AlertService = Ng4AlertService;
exports.Ng4AlertComponent = Ng4AlertComponent;

Object.defineProperty(exports, '__esModule', { value: true });

})));
