// aws-amplify-angular/lib/components/authenticator/confirm-sign-in-component/confirm-sign-in-component.core.js
import { Component, Input } from '@angular/core';
import { AmplifyService } from '../../../providers';
var template = "\n<div class=\"amplify-form-container\" *ngIf=\"_show\">\n  <div class=\"amplify-form-body\">\n\n    <div class=\"amplify-form-row\">\n      <div class=\"amplify-form-cell-left\">\n        <a class=\"amplify-form-link\"\n          (click)=\"onSignIn()\"\n        >Back to Sign In</a>\n      </div>\n    </div>\n\n    <div class=\"amplify-form-row\">\n      <input #code\n        (keyup)=\"setCode(code.value)\"\n        (keyup.enter)=\"onConfirm()\"\n        class=\"amplify-form-input\"\n        type=\"text\"\n        placeholder=\"Code\"\n      />\n    </div>\n    <button class=\"amplify-form-button\"\n      (click)=\"onConfirm()\"\n    >Confirm</button>\n  </div>\n  <div class=\"amplify-form-footer\">\n    <div class=\"amplify-form-message-error\" *ngIf=\"errorMessage\">{{ errorMessage }}</div>\n  </div>\n</div>\n";
var ConfirmSignInComponentCore = /** @class */ (function () {
    function ConfirmSignInComponentCore(amplifyService) {
        this.amplifyService = amplifyService;
    }
    Object.defineProperty(ConfirmSignInComponentCore.prototype, "data", {
        set: function (data) {
            this._authState = data.authState;
            this._show = data.authState.state === 'confirmSignIn';
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ConfirmSignInComponentCore.prototype, "authState", {
        set: function (authState) {
            this._authState = authState;
            this._show = authState.state === 'confirmSignIn';
        },
        enumerable: true,
        configurable: true
    });
    ConfirmSignInComponentCore.prototype.setCode = function (code) {
        this.code = code;
    };
    ConfirmSignInComponentCore.prototype.onConfirm = function () {
        var _this = this;
        var user = this._authState.user;
        var challengeName = user.challengeName;
        var mfaType = challengeName === 'SOFTWARE_TOKEN_MFA' ? challengeName : null;
        this.amplifyService.auth()
            .confirmSignIn(user, this.code, mfaType)
            .then(function () {
            _this.amplifyService.setAuthState({ state: 'signedIn', user: user });
        })
            .catch(function (err) { return _this._setError(err); });
    };
    ConfirmSignInComponentCore.prototype.onSignIn = function () {
        this.amplifyService.setAuthState({ state: 'signIn', user: null });
    };
    ConfirmSignInComponentCore.prototype._setError = function (err) {
        if (!err) {
            this.errorMessage = null;
            return;
        }
        this.errorMessage = err.message || err;
    };
    ConfirmSignInComponentCore.decorators = [
        { type: Component, args: [{
                    selector: 'amplify-auth-confirm-sign-in-core',
                    template: template
                },] },
    ];
    /** @nocollapse */
    ConfirmSignInComponentCore.ctorParameters = function () { return [
        { type: AmplifyService, },
    ]; };
    ConfirmSignInComponentCore.propDecorators = {
        "data": [{ type: Input },],
        "authState": [{ type: Input },],
    };
    return ConfirmSignInComponentCore;
}());
export { ConfirmSignInComponentCore };
