// ng2-alert-service/dist/toast-message-component/toast-message-component.component.js
import { Component, Input } from '@angular/core';
var ToastMessageComponentComponent = /** @class */ (function () {
    function ToastMessageComponentComponent() {
        this.alerts = [];
    }
    ToastMessageComponentComponent.prototype.ngOnInit = function () {
    };
    ToastMessageComponentComponent.prototype.closeAlert = function (alert) {
        this.alerts.splice(this.alerts.indexOf(alert), 1);
    };
    ToastMessageComponentComponent.decorators = [
        { type: Component, args: [{
                    selector: 'toast-message-component',
                    styles: ["div.floating-alert{\n                position: fixed;\n                right: 1%;\n                top: 1em;\n                z-index: 10;\n              button{\n                  margin-left: 20px;\n              }\n            }"],
                    template: "<div class=\"floating-alert\">\n                <div class=\"alert alert-{{alertObject.type}}\" *ngFor=\"let alertObject of alerts\">\n                  <button class=\"close\" data-dismiss=\"alert\" aria-label=\"close\" title=\"close\" (click)=\"closeAlert(alertObject)\">  <span aria-hidden=\"true\">&times;</span></button>    {{alertObject.message}}\n                </div>\n              </div>"
                },] },
    ];
    /** @nocollapse */
    ToastMessageComponentComponent.ctorParameters = function () { return []; };
    ToastMessageComponentComponent.propDecorators = {
        "alerts": [{ type: Input },],
    };
    return ToastMessageComponentComponent;
}());
export { ToastMessageComponentComponent };
