// aws-amplify-angular/lib/components/authenticator/greeting-component/greeting.component.core.js
import { Component, Input } from '@angular/core';
import { AmplifyService } from '../../../providers';
var template = "\n<div class=\"amplify-greeting\">\n    <span class=\"amplify-greeting-text\">{{ greeting }}</span>\n    <button class=\"amplify-form-button amplify-greeting-sign-out\"\n        *ngIf=\"signedIn\"\n        (click)=\"onSignOut()\"\n      >Sign Out</button>\n</div>\n";
var GreetingComponentCore = /** @class */ (function () {
    function GreetingComponentCore(amplifyService) {
        this.amplifyService = amplifyService;
        this.subscribe();
    }
    GreetingComponentCore.prototype.subscribe = function () {
        var _this = this;
        this.amplifyService.authStateChange$
            .subscribe(function (state) { return _this.setAuthState(state); });
    };
    GreetingComponentCore.prototype.setAuthState = function (authState) {
        this.authState = authState;
        this.signedIn = authState.state === 'signedIn';
        this.greeting = this.signedIn
            ? "Hello " + authState.user.username
            : "";
    };
    GreetingComponentCore.prototype.onSignOut = function () {
        this.amplifyService.auth().signOut();
    };
    GreetingComponentCore.decorators = [
        { type: Component, args: [{
                    selector: 'amplify-auth-greetings-core',
                    template: template
                },] },
    ];
    /** @nocollapse */
    GreetingComponentCore.ctorParameters = function () { return [
        { type: AmplifyService, },
    ]; };
    GreetingComponentCore.propDecorators = {
        "authState": [{ type: Input },],
    };
    return GreetingComponentCore;
}());
export { GreetingComponentCore };
