// aws-amplify-angular/lib/components/authenticator/forgot-password-component/forgot-password.component.core.js
import { Component, Input } from '@angular/core';
import { AmplifyService } from '../../../providers';
var template = "\n<div class=\"amplify-form-container\" *ngIf=\"_show\">\n  <div class=\"amplify-form-body\">\n\n    <div class=\"amplify-form-row\">\n      \n      <div class=\"amplify-form-cell-left\">\n        <a class=\"amplify-form-link\"\n          (click)=\"onSignIn()\"\n        >Back to Sign In</a>\n      </div>\n\n    </div>\n\n    <div class=\"amplify-form-row\">\n      <input\n        (keyup)=\"setUsername($event.target.value)\"\n        class=\"amplify-form-input\"\n        type=\"text\"\n        placeholder=\"Username\"\n        [value]=\"username\"\n      />\n    </div>\n    <div class=\"amplify-form-row\">\n      <input #code\n        (keyup)=\"setCode(code.value)\"\n        class=\"amplify-form-input\"\n        type=\"text\"\n        placeholder=\"Code\"\n      />\n    </div>\n    <div class=\"amplify-form-row\">\n      <input #password\n        (keyup)=\"setPassword(password.value)\"\n        (keyup.enter)=\"onSubmit()\"\n        class=\"amplify-form-input\"\n        type=\"password\"\n        placeholder=\"Password\"\n      />\n    </div>\n    <div class=\"amplify-form-row\">\n    <button class=\"amplify-form-button\"\n      (click)=\"onSend()\"\n    >Send Code</button>\n    <button class=\"amplify-form-button\"\n      (click)=\"onSubmit()\"\n    >Submit</button>\n    </div>\n  </div>\n  <div class=\"amplify-form-footer\">\n    <div class=\"amplify-form-message-error\" *ngIf=\"errorMessage\">{{ errorMessage }}</div>\n  </div>\n</div>\n";
var ForgotPasswordComponentCore = /** @class */ (function () {
    function ForgotPasswordComponentCore(amplifyService) {
        this.amplifyService = amplifyService;
    }
    Object.defineProperty(ForgotPasswordComponentCore.prototype, "data", {
        set: function (data) {
            this._authState = data.authState;
            this._show = data.authState.state === 'forgotPassword';
            this.username = data.authState.user ? data.authState.user.username || '' : '';
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ForgotPasswordComponentCore.prototype, "authState", {
        set: function (authState) {
            this._authState = authState;
            this._show = authState.state === 'forgotPassword';
            this.username = authState.user ? authState.user.username || '' : '';
        },
        enumerable: true,
        configurable: true
    });
    ForgotPasswordComponentCore.prototype.setUsername = function (username) {
        this.username = username;
    };
    ForgotPasswordComponentCore.prototype.setCode = function (code) {
        this.code = code;
    };
    ForgotPasswordComponentCore.prototype.setPassword = function (password) {
        this.password = password;
    };
    ForgotPasswordComponentCore.prototype.onSend = function () {
        var _this = this;
        this.amplifyService.auth().forgotPassword(this.username)
            .then(function () { return console.log('code sent'); })
            .catch(function (err) { return _this._setError(err); });
    };
    ForgotPasswordComponentCore.prototype.onSubmit = function () {
        var _this = this;
        this.amplifyService.auth()
            .forgotPasswordSubmit(this.username, this.code, this.password)
            .then(function () {
            var user = { username: _this.username };
            _this.amplifyService.setAuthState({ state: 'signIn', user: user });
        })
            .catch(function (err) { return _this._setError(err); });
    };
    ForgotPasswordComponentCore.prototype.onSignIn = function () {
        this.amplifyService.setAuthState({ state: 'signIn', user: null });
    };
    ForgotPasswordComponentCore.prototype._setError = function (err) {
        if (!err) {
            this.errorMessage = null;
            return;
        }
        this.errorMessage = err.message || err;
    };
    ForgotPasswordComponentCore.decorators = [
        { type: Component, args: [{
                    selector: 'amplify-auth-forgot-password-core',
                    template: template
                },] },
    ];
    /** @nocollapse */
    ForgotPasswordComponentCore.ctorParameters = function () { return [
        { type: AmplifyService, },
    ]; };
    ForgotPasswordComponentCore.propDecorators = {
        "data": [{ type: Input },],
        "authState": [{ type: Input },],
    };
    return ForgotPasswordComponentCore;
}());
export { ForgotPasswordComponentCore };
