// aws-amplify-angular/lib/components/authenticator/sign-up-component/sign-up.component.core.js
import { Component, Input } from '@angular/core';
import { AmplifyService } from '../../../providers';
var template = "\n<div class=\"amplify-form-container\" *ngIf=\"_show\">\n  <div class=\"amplify-form-body\">\n\n    <div class=\"amplify-form-row\">\n\n      <div class=\"amplify-form-cell-left\">\n        <a class=\"amplify-form-link\"\n          (click)=\"onSignIn()\"\n        >Sign In</a>\n      </div>\n\n      <div class=\"amplify-form-cell-right\">\n        <a class=\"amplify-form-link\"\n          (click)=\"onSignUp()\"\n        >Sign Up</a>\n      </div>\n\n    </div>\n\n    <div class=\"amplify-form-row\">\n      <input #username\n        (keyup)=\"setUsername(username.value)\"\n        class=\"amplify-form-input\"\n        type=\"text\"\n        placeholder=\"Username\"\n      />\n    </div>\n    <div class=\"amplify-form-row\">\n      <input #password\n        (keyup)=\"setPassword(password.value)\"\n        (keyup.enter)=\"onSignUp()\"\n        class=\"amplify-form-input\"\n        type=\"password\"\n        placeholder=\"Password\"\n      />\n    </div>\n    <div class=\"amplify-form-row\">\n      <input #email\n        (keyup)=\"setEmail(email.value)\"\n        class=\"amplify-form-input\"\n        type=\"text\"\n        placeholder=\"Email\"\n      />\n    </div>\n    <div class=\"amplify-form-row\">\n      <input #phone_number\n        (keyup)=\"setPhoneNumber(phone_number.value)\"\n        class=\"amplify-form-input\"\n        type=\"text\"\n        placeholder=\"Phone Number\"\n      />\n    </div>\n    <div class=\"amplify-form-row\">\n      <div class=\"amplify-form-cell-right\">\n        <a class=\"amplify-form-link\"\n          (click)=\"onConfirmSignUp()\"\n        >Confirm a Code</a>\n      </div>\n    </div>\n    <button class=\"amplify-form-button\"\n      (click)=\"onSignUp()\"\n    >Sign Up</button>\n  </div>\n  <div class=\"amplify-form-footer\">\n    <div class=\"amplify-form-message-error\" *ngIf=\"errorMessage\">{{ errorMessage }}</div>\n  </div>\n</div>\n";
var SignUpComponentCore = /** @class */ (function () {
    function SignUpComponentCore(amplifyService) {
        this.amplifyService = amplifyService;
    }
    Object.defineProperty(SignUpComponentCore.prototype, "authState", {
        set: function (authState) {
            this._authState = authState;
            this._show = authState.state === 'signUp';
        },
        enumerable: true,
        configurable: true
    });
    SignUpComponentCore.prototype.setUsername = function (username) {
        this.username = username;
    };
    SignUpComponentCore.prototype.setPassword = function (password) {
        this.password = password;
    };
    SignUpComponentCore.prototype.setEmail = function (email) {
        this.email = email;
    };
    SignUpComponentCore.prototype.setPhoneNumber = function (phone_number) {
        this.phone_number = phone_number;
    };
    SignUpComponentCore.prototype.onSignUp = function () {
        var _this = this;
        this.amplifyService.auth()
            .signUp(this.username, this.password, this.email, this.phone_number)
            .then(function (user) { return _this.amplifyService.setAuthState({ state: 'confirmSignUp', user: null }); })
            .catch(function (err) { return _this._setError(err); });
    };
    SignUpComponentCore.prototype.onSignIn = function () {
        this.amplifyService.setAuthState({ state: 'signIn', user: null });
    };
    SignUpComponentCore.prototype.onConfirmSignUp = function () {
        this.amplifyService.setAuthState({ state: 'confirmSignUp', user: null });
    };
    SignUpComponentCore.prototype._setError = function (err) {
        if (!err) {
            this.errorMessage = null;
            return;
        }
        this.errorMessage = err.message || err;
    };
    SignUpComponentCore.decorators = [
        { type: Component, args: [{
                    selector: 'amplify-auth-sign-up-core',
                    template: template
                },] },
    ];
    /** @nocollapse */
    SignUpComponentCore.ctorParameters = function () { return [
        { type: AmplifyService, },
    ]; };
    SignUpComponentCore.propDecorators = {
        "authState": [{ type: Input },],
    };
    return SignUpComponentCore;
}());
export { SignUpComponentCore };
