// aws-amplify-angular/lib/components/storage/photo-picker-component/photo-picker.component.core.js
import { Component, Input, Output, EventEmitter } from '@angular/core';
var template = "\n<div class=\"amplify-photo-picker-container\">\n  <div class=\"amplify-photo-picker-preview\">\n    <img\n      class=\"amplify-photo-picker-preview\"\n      src=\"{{photoUrl}}\"\n      *ngIf=\"hasPhoto\"\n      (error)=\"onPhotoError()\"\n    />\n  </div>\n  <div>\n    <input type=\"file\" \n      class=\"amplify-upload-input\"\n      accept=\"image/*\"\n      (change)=\"pick($event)\"/>\n  </div>\n\n</div>\n";
var PhotoPickerComponentCore = /** @class */ (function () {
    function PhotoPickerComponentCore() {
        this.hasPhoto = false;
        this.picked = new EventEmitter();
        this.loaded = new EventEmitter();
    }
    Object.defineProperty(PhotoPickerComponentCore.prototype, "url", {
        set: function (url) {
            this.photoUrl = url;
            this.hasPhoto = true;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(PhotoPickerComponentCore.prototype, "data", {
        set: function (data) {
            this.photoUrl = data.url;
            this.hasPhoto = true;
        },
        enumerable: true,
        configurable: true
    });
    PhotoPickerComponentCore.prototype.pick = function (evt) {
        var file = evt.target.files[0];
        if (!file) {
            return;
        }
        var name = file.name, size = file.size, type = file.type;
        this.picked.emit(file);
        var that = this;
        var reader = new FileReader();
        reader.onload = function (e) {
            var target = e.target;
            var url = target.result;
            that.photoUrl = url;
            that.hasPhoto = true;
            that.loaded.emit(url);
        };
        reader.readAsDataURL(file);
    };
    PhotoPickerComponentCore.prototype.onPhotoError = function () {
        this.hasPhoto = false;
    };
    PhotoPickerComponentCore.decorators = [
        { type: Component, args: [{
                    selector: 'amplify-photo-picker-core',
                    template: template
                },] },
    ];
    /** @nocollapse */
    PhotoPickerComponentCore.propDecorators = {
        "url": [{ type: Input },],
        "data": [{ type: Input },],
        "picked": [{ type: Output },],
        "loaded": [{ type: Output },],
    };
    return PhotoPickerComponentCore;
}());
export { PhotoPickerComponentCore };
