// aws-amplify-angular/lib/components/authenticator/confirm-sign-up-component/confirm-sign-up.component.core.js
import { Component, Input } from '@angular/core';
import { AmplifyService } from '../../../providers';
var template = "\n<div class=\"amplify-form-container\" *ngIf=\"_show\">\n  <div class=\"amplify-form-body\">\n\n    <div class=\"amplify-form-row\">\n      \n      <div class=\"amplify-form-cell-left\">\n        <a class=\"amplify-form-link\"\n          (click)=\"onSignIn()\"\n        >Back to Sign In</a>\n      </div>\n\n    </div>\n\n    <div class=\"amplify-form-row\">\n      <input\n        (keyup)=\"setUsername($event.target.value)\"\n        class=\"amplify-form-input\"\n        type=\"text\"\n        placeholder=\"Username\"\n        [value]=\"username\"\n      />\n    </div>\n    <div class=\"amplify-form-row\">\n      <input #code\n        (keyup)=\"setCode(code.value)\"\n        (keyup.enter)=\"onConfirm()\"\n        class=\"amplify-form-input\"\n        type=\"text\"\n        placeholder=\"Code\"\n      />\n    </div>\n      \n    <button class=\"amplify-form-button\"\n      (click)=\"onConfirm()\">Confirm</button>\n    <button class=\"amplify-form-button\"\n      (click)=\"onResend()\">Resend</button>\n\n  </div>\n\n  <div class=\"amplify-form-footer\">\n    <div class=\"amplify-form-message-error\" *ngIf=\"errorMessage\">{{ errorMessage }}</div>\n  </div>\n\n</div>\n";
var ConfirmSignUpComponentCore = /** @class */ (function () {
    function ConfirmSignUpComponentCore(amplifyService) {
        this.amplifyService = amplifyService;
    }
    Object.defineProperty(ConfirmSignUpComponentCore.prototype, "data", {
        set: function (data) {
            this._authState = data.authState;
            this._show = data.authState.state === 'confirmSignUp';
            this.username = data.authState.user ? data.authState.user.username || '' : '';
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ConfirmSignUpComponentCore.prototype, "authState", {
        set: function (authState) {
            this._authState = authState;
            this._show = authState.state === 'confirmSignUp';
            this.username = authState.user ? authState.user.username || '' : '';
        },
        enumerable: true,
        configurable: true
    });
    ConfirmSignUpComponentCore.prototype.setUsername = function (username) {
        this.username = username;
    };
    ConfirmSignUpComponentCore.prototype.setCode = function (code) {
        this.code = code;
    };
    ConfirmSignUpComponentCore.prototype.onConfirm = function () {
        var _this = this;
        this.amplifyService.auth()
            .confirmSignUp(this.username, this.code)
            .then(function () { return console.log('confirm success'); })
            .catch(function (err) { return _this._setError(err); });
    };
    ConfirmSignUpComponentCore.prototype.onResend = function () {
        var _this = this;
        this.amplifyService.auth().resendSignUp(this.username)
            .then(function () { return console.log('code resent'); })
            .catch(function (err) { return _this._setError(err); });
    };
    ConfirmSignUpComponentCore.prototype.onSignIn = function () {
        this.amplifyService.setAuthState({ state: 'signIn', user: null });
    };
    ConfirmSignUpComponentCore.prototype._setError = function (err) {
        if (!err) {
            this.errorMessage = null;
            return;
        }
        this.errorMessage = err.message || err;
    };
    ConfirmSignUpComponentCore.decorators = [
        { type: Component, args: [{
                    selector: 'amplify-auth-confirm-sign-up-core',
                    template: template
                },] },
    ];
    /** @nocollapse */
    ConfirmSignUpComponentCore.ctorParameters = function () { return [
        { type: AmplifyService, },
    ]; };
    ConfirmSignUpComponentCore.propDecorators = {
        "data": [{ type: Input },],
        "authState": [{ type: Input },],
    };
    return ConfirmSignUpComponentCore;
}());
export { ConfirmSignUpComponentCore };
