// aws-amplify-angular/lib/aws-amplify-ionic-module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { AmplifyService } from './providers/amplify.service';
import * as components from './components';
import { IonicModule } from '@ionic/angular';
var exportables = [
    components.AuthenticatorIonicComponent,
    components.ConfirmSignInComponentIonic,
    components.ConfirmSignUpComponentIonic,
    components.ForgotPasswordComponentIonic,
    components.GreetingComponentIonic,
    components.PhotoPickerIonicComponent,
    components.S3AlbumComponentIonic,
    components.S3ImageComponentIonic,
    components.SignInComponentIonic,
    components.SignUpComponentIonic,
    components.ChatbotComponentIonic,
    components.RequireNewPasswordComponentIonic
];
var AmplifyIonicModule = /** @class */ (function () {
    function AmplifyIonicModule() {
    }
    AmplifyIonicModule.decorators = [
        { type: NgModule, args: [{
                    imports: [
                        CommonModule,
                        IonicModule
                    ],
                    declarations: exportables.slice(),
                    entryComponents: exportables.slice(),
                    providers: [AmplifyService],
                    exports: exportables.slice()
                },] },
    ];
    return AmplifyIonicModule;
}());
export { AmplifyIonicModule };
