// aws-amplify-angular/lib/components/authenticator/sign-in-component/sign-in.component.core.js
import { Component, Input } from '@angular/core';
import { AmplifyService } from '../../../providers';
import { includes } from '../common';
var template = "\n<div class=\"amplify-form-container\" *ngIf=\"_show\">\n  <div class=\"amplify-form-header\"></div>\n  <div class=\"amplify-form-body\">\n    \n    <div class=\"amplify-form-row\">\n      \n      <div class=\"amplify-form-cell-left\">\n        <a class=\"amplify-form-link\"\n          (click)=\"onSignIn()\"\n        >Sign In</a>\n      </div>\n      \n      <div class=\"amplify-form-cell-right\">\n        <a class=\"amplify-form-link\"\n          (click)=\"onSignUp()\"\n        >Sign Up</a>\n      </div>\n\n    </div>\n\n    <div class=\"amplify-form-row\">\n      <input\n        (keyup)=\"setUsername($event.target.value)\"\n        class=\"amplify-form-input\"\n        type=\"text\"\n        placeholder=\"Username\"\n        [value]=\"username\"\n      />\n    </div>\n\n    <div class=\"amplify-form-row\">\n      <input #password\n        (keyup)=\"setPassword(password.value)\"\n        (keyup.enter)=\"onSignIn()\"\n        class=\"amplify-form-input\"\n        type=\"password\"\n        placeholder=\"Password\"\n      />\n    </div>\n\n    <div class=\"amplify-form-row\">\n\n      <div class=\"amplify-form-cell-right\">\n        <a class=\"amplify-form-link\"\n          (click)=\"onForgotPassword()\"\n        >Forgot Password</a>\n      </div>\n\n    </div>\n\n    <button class=\"amplify-form-button\"\n      (click)=\"onSignIn()\"\n    >Sign In</button>\n\n  </div>\n\n  <div class=\"amplify-form-footer\">\n    <div class=\"amplify-form-message-error\" *ngIf=\"errorMessage\">{{ errorMessage }}</div>\n  </div>\n\n</div>\n";
var SignInComponentCore = /** @class */ (function () {
    function SignInComponentCore(amplifyService) {
        this.amplifyService = amplifyService;
    }
    Object.defineProperty(SignInComponentCore.prototype, "authState", {
        set: function (authState) {
            this._authState = authState;
            this._show = includes(['signIn', 'signedOut', 'signedUp'], authState.state);
            this.username = authState.user ? authState.user.username || '' : '';
        },
        enumerable: true,
        configurable: true
    });
    SignInComponentCore.prototype.setUsername = function (username) {
        this.username = username;
    };
    SignInComponentCore.prototype.setPassword = function (password) {
        this.password = password;
    };
    SignInComponentCore.prototype.onSignIn = function () {
        var _this = this;
        this.amplifyService.auth().signIn(this.username, this.password)
            .then(function (user) {
            if (user.challengeName === 'SMS_MFA' || user.challengeName === 'SOFTWARE_TOKEN_MFA') {
                _this.amplifyService.setAuthState({ state: 'confirmSignIn', user: user });
            }
            else if (user.challengeName === 'NEW_PASSWORD_REQUIRED') {
                _this.amplifyService.setAuthState({ state: 'requireNewPassword', user: user });
            }
            else {
                _this.amplifyService.setAuthState({ state: 'signedIn', user: user });
            }
        })
            .catch(function (err) { return _this._setError(err); });
    };
    SignInComponentCore.prototype.onForgotPassword = function () {
        var user = this.username ? { username: this.username } : null;
        this.amplifyService.setAuthState({ state: 'forgotPassword', user: user });
    };
    SignInComponentCore.prototype.onSignUp = function () {
        var user = this.username ? { username: this.username } : null;
        this.amplifyService.setAuthState({ state: 'signUp', user: user });
    };
    SignInComponentCore.prototype._setError = function (err) {
        if (!err) {
            this.errorMessage = null;
            return;
        }
        this.errorMessage = err.message || err;
    };
    SignInComponentCore.decorators = [
        { type: Component, args: [{
                    selector: 'amplify-auth-sign-in-core',
                    template: template
                },] },
    ];
    /** @nocollapse */
    SignInComponentCore.ctorParameters = function () { return [
        { type: AmplifyService, },
    ]; };
    SignInComponentCore.propDecorators = {
        "authState": [{ type: Input },],
    };
    return SignInComponentCore;
}());
export { SignInComponentCore };
